packages/core/src/auth/providers/sharedCredentialsProviderFactory.ts

```diff
@@ -3,47 +3,23 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import fs from '../../shared/fs/fs'
 import { getLogger, Logger } from '../../shared/logger'
 import { loadSharedCredentialsSections, updateAwsSdkLoadConfigEnvVar } from '../credentials/sharedCredentials'
 import { CredentialsProviderType } from './credentials'
 import { BaseCredentialsProviderFactory } from './credentialsProviderFactory'
 import { SharedCredentialsProvider } from './sharedCredentialsProvider'
-import { getCredentialsFilename, getConfigFilename } from '../credentials/sharedCredentialsFile'
 
 export class SharedCredentialsProviderFactory extends BaseCredentialsProviderFactory<SharedCredentialsProvider> {
     private readonly logger: Logger = getLogger()
 
-    private loadedCredentialsModificationMillis?: number
-    private loadedConfigModificationMillis?: number
-
     public async refresh(): Promise<void> {
-        if (await this.needsRefresh()) {
-            await this.loadSharedCredentialsProviders()
-        }
+        await this.loadSharedCredentialsProviders()
     }
 
     public override getProviderType(): CredentialsProviderType | undefined {
         return SharedCredentialsProvider.getProviderType()
     }
 
-    protected override resetProviders() {
-        this.loadedCredentialsModificationMillis = undefined
-        this.loadedConfigModificationMillis = undefined
-
-        super.resetProviders()
-    }
-
-    private async needsRefresh(): Promise<boolean> {
-        const credentialsLastModMillis = await this.getLastModifiedMillis(getCredentialsFilename())
-        const configLastModMillis = await this.getLastModifiedMillis(getConfigFilename())
-
-        return (
-            this.loadedCredentialsModificationMillis !== credentialsLastModMillis ||
-            this.loadedConfigModificationMillis !== configLastModMillis
```

Comment:
Can't remember if this was discussed already: maybe the delete/modification in the test happened too quickly, so the mtime didn't actually change on the re-created file?

Reply:
Thats what I suspect as well. When I added a timeout in-between the two, the test appeared to pass as well.